File: benchmarks/pandas/bench_transform_agg.py
"""Benchmark: Series.transform — transform a 100k-element Series"""
import json, time
import numpy as np
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

data = (np.arange(ROWS) % 500) + 1.0
idx = np.arange(ROWS) % 500
s = pd.Series(data, index=idx)

for _ in range(WARMUP):
    s.transform("mean")

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.transform("mean")
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "transform_agg",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_transform_agg.ts
/**
 * Benchmark: seriesTransform — transform a 100k-element Series
 */
import { Series, seriesTransform } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const data = Float64Array.from({ length: ROWS }, (_, i) => (i % 500) + 1);
const s = new Series({ data, index: Array.from({ length: ROWS }, (_, i) => i % 500) });

for (let i = 0; i < WARMUP; i++) {
  seriesTransform(s, "mean");
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  seriesTransform(s, "mean");
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "transform_agg",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
